src/async-wrap.h

```diff
@@ -157,6 +157,9 @@ class AsyncWrap : public BaseObject {
 
   virtual size_t self_size() const = 0;
 
+  class DestroyParam;
```

Comment:
I think we either have to leave this public, or move RegisterDestroyHook into `AsyncWrap`. Since the other registered methods don't do that, I wasn't sure if that'd work?

Reply:
Without having looked too much, you can assume that any code behind `NODE_WANT_INTERNALS` (including this header file) is not considered public